external-crates/move/crates/move-compiler/tests/sui_mode/linter/random_objects.move

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module a::test1 {
+    use sui::random::{Random, RandomGenerator};
+
+    entry fun should_work(_r: &Random) {}
+
+    public fun not_allowed1(_x: u64, _r: &Random) {}
+    public fun not_allowed2(_rg: &RandomGenerator, _x: u64) {}
+    public fun not_allowed3(_r: &Random, _rg: &RandomGenerator, _x: u64) {}
+
+    #[allow(lint(random_objects))]
+    public fun allow_random_objects_should_work(_r: &Random, _rg: &RandomGenerator) {}
```

Comment:
Nit, I would make this a separate test

Reply:
all linter related tests use this pattern (all tests in the same file)